test/conformance/walk.go

```diff
@@ -14,415 +14,294 @@ See the License for the specific language governing permissions and
-	tokens := []string{}
-	for _, describe := range v.describes {
-		tokens = append(tokens, describe.text)
-		if len(describe.lastContext.text) > 0 {
-			tokens = append(tokens, describe.lastContext.text)
+		for _, data := range dataSet {
+			fmt.Printf("## [%s](%s)\n\n", data.TestName, data.URL)
+			fmt.Printf("- Added to conformance in release %s\n", data.Release)
+			fmt.Printf("- Defined in code as: %s\n\n", data.CodeName)
+			fmt.Printf("%s\n\n", data.Description)
 		}
+		return
 	}
-	tokens = append(tokens, value)
 
-	trimmed := []string{}
-	for _, token := range tokens {
-		trimmed = append(trimmed, strings.Trim(token, "\""))
+	// Serialize the list as a whole. Generally meant to end up as conformance.txt which tracks the set of tests.
+	b, err := yaml.Marshal(dataSet)
+	if err != nil {
+		log.Printf("Error marshalling data into YAML: %v", err)
 	}
-
-	return strings.Join(trimmed, " ")
+	fmt.Println(string(b))
 }
 
-var (
-	regexTag = regexp.MustCompile(`(\[[a-zA-Z0-9:-]+\])`)
-)
-
-// normalizeTestName removes tags (e.g., [Feature:Foo]), double quotes and trim
-// the spaces to normalize the test name.
-func normalizeTestName(s string) string {
-	r := regexTag.ReplaceAllString(s, "")
-	r = strings.Trim(r, "\"")
-	return strings.TrimSpace(r)
-}
-
-func validateTestName(s string) error {
-	matches := regexIneligibleTags.FindAllString(s, -1)
-	if matches != nil {
-		return fmt.Errorf("'%s' cannot have invalid tags %v", s, strings.Join(matches, ","))
+// TODO(SCHNAKE) TEST if I split this up to first get the comment map then pass comment map/stack trace to pick out the right one that would be valuable
+func getConformanceDataFromStackTrace(fullstackstrace string) (*conformanceData, error) {
+	// The full stacktrace to parse from ginkgo is of the form:
+	// k8s.io/kubernetes/test/e2e/storage/utils.SIGDescribe(0x51f4c4f, 0xf, 0x53a0dd8, 0xc000ab6e01)\n\ttest/e2e/storage/utils/framework.go:23 +0x75\n ... ...
+	// So we need to split it into lines, remove whitespace, and then grab the files/lines.
+	stack := strings.Replace(fullstackstrace, "\t", "", -1)
+	calls := strings.Split(stack, "\n")
+	frames := []frame{}
+	i := 0
+	for i < len(calls) {
+		fileLine := strings.Split(calls[i+1], " ")
+		lineinfo := strings.Split(fileLine[0], ":")
+		line, err := strconv.Atoi(lineinfo[1])
+		if err != nil {
+			panic(err)
+		}
+		frames = append(frames, frame{
+			Function: calls[i],
+			File:     lineinfo[0],
+			Line:     line,
+		})
+		i += 2
 	}
-	return nil
-}
 
-// funcName converts a selectorExpr with two idents into a string,
-// x.y -> "x.y"
-func funcName(n ast.Expr) string {
-	if sel, ok := n.(*ast.SelectorExpr); ok {
-		if x, ok := sel.X.(*ast.Ident); ok {
-			return x.String() + "." + sel.Sel.String()
-		}
+	// filenames have `/go/src/k8s.io` prefix which dont exist locally
+	for i, v := range frames {
+		frames[i].File = strings.Replace(v.File,
+			"/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes",
```

Comment:
I need some additional 👀 on this part; this was hacked together based on my own work but I'm not sure if I can always expect the same prefix or what I should be using to strip this prefix as appropriate.

Reply:
I will try to get this running locally and report back

Reply:
@spiffxp did you try this? It looks fishy.